source/common/http/conn_manager_impl.h

```diff
@@ -193,10 +193,10 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void decodeTrailers(RequestTrailerMapPtr&& trailers) override;
 
     // Tracing::TracingConfig
-    Tracing::OperationName operationName() const override;
-    const Tracing::CustomTagMap* customTags() const override;
-    bool verbose() const override;
-    uint32_t maxPathTagLength() const override;
+    absl::optional<Tracing::OperationName> operationName() const override;
+    const absl::optional<Tracing::CustomTagMap*> customTags() const override;
+    absl::optional<bool> verbose() const override;
+    absl::optional<uint32_t> maxPathTagLength() const override;
```

Comment:
Sorry I think I may have lead you astray here. I just looked at all of this more closely.

Instead of having `ActiveStream` implement TracingConfig, can you just instead return the actual tracing config (or not) from `tracingConfig()`? Then I think you can just revert all of the other interface changes and this will be simpler?